### DISCUSSION TITLE
V4: How to pre-select vertical from URL parameter?

### DISCUSSION THREAD

**Discussion by nikosmsk** [microsoft-search/pnp-modern-search#2079] [Q&A] [answered]
**Is your feature request related to a problem? Please describe.**
Is it possible to pre-select a vertical based on a parameter from URL in V4? So that a certain vertical item gets selected and respective search results are shown when the page loads.

**Describe alternatives you've considered**
1) Use V3 search web parts or SP classic search with search navigation web part 
2) Multiple modern search pages, but layoutType HeaderlessSearchResults has limitation of one page per site collection  

**Additional context**
Users should see relevant results when they access the search results page from particular places in Dynamics365 CE or F&O. For example, to see all sales documents for customer X or all purchase documents for vendor Y (customer ID and vendor ID are metadata in various document libraries and "Purchase" and "Sales" are the verticals)

**Comment by wobba** [comment]
https://microsoft-search.github.io/pnp-modern-search/

**Comment by nikosmsk** [comment]
Hi @wobba , sorry to be not precise enough. In V3 there was a property "Default vertical querystring parameter",  which is not in V4 (yet?) . Also the filter values from URL ( ?filters=[{"n":"FileType","t":["docx"]}] )  have no effect in V4 ...

**Comment by wobba** [comment]
Correct, not in v4 right now. Sorry for not reading it properly 🙂

**Comment by bryllyg** [comment]
Putting my vote in for this enhancement. We have it set up in V3, but are migrating to V4 and would like to maintain the same tab behavior.

**Comment by SjoerdV** [comment]
+1

**Comment by FranckyC** [comment, marked as answer]
In v4 you can une the `v` parameter with tab name to preselect it.

**Reply by wobba** [reply]
Maybe we should add that to https://microsoft-search.github.io/pnp-modern-search/usage/search-verticals/ ?

**Reply by SjoerdV** [reply]
Is it correct, this does not work if the Vertical tab has a space in the name? I couldn't make this work in any case, maybe there's a trick to it?

**Reply by FranckyC** [reply]
This should work wit hspaces. You can alos escape space with %20 in the URL

**Reply by adam-pope-arup** [reply]
Great to hear this works. I've tried the following:

..../search.aspx#v=TabName
..../search.aspx#v:TabName
..../search.aspx?v=TabName
..../search.aspx?v:TabName
..../search.aspx?vTabName
..../search.aspx#vTabName
..../search.aspx?vTabName
.../search.aspx#v=(TabName)
.../search.aspx?v=(TabName)

...without success, unfortunately. My configuration does work with .../search.aspx#keyword or .../search.aspx##RefinableString123:term as URL parameters though.

As @wobba  suggests, perhaps some documentation would help people get this working? Thank you!
